Update user-assigned and system-assigned managed identity code, overview edits (#44872)
dotnet/docs: docs/azure/sdk/includes/implement-managed-identity-concepts.md

---
ms.topic: include
ms.date: 02/12/2025
---

## Authenticate to Azure services from your app

> To authenticate to Azure services from your app code using Microsoft Entra ID and a managed identity, use an implementation of the [`TokenCredential`](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet#credentials) class. The [Azure.Identity](/dotnet/api/azure.identity) library provides various implementations of `TokenCredential` for different scenarios and credential types. These classes allow apps to seamlessly authenticate to Azure resources whether the app is running locally, deployed to Azure, or on an on-premises server.

Reviewer: If it's managed identity that's being used, then we should say `ManagedIdentityCredential` instead of "an implementation of TokenCredential".

> The [Azure.Identity](/dotnet/api/azure.identity) library provides various implementations of `TokenCredential` for different scenarios and credential types. To authenticate to Azure services from your app code using Microsoft Entra ID and a managed identity, use the [`ManagedIdentityCredential`](/dotnet/api/azure.identity.managedidentitycredential?view=azure-dotnet) class. These classes allow apps to seamlessly authenticate to Azure resources whether the app is running locally, deployed to Azure, or on an on-premises server.
> 
> The steps ahead demonstrate how to use a `TokenCredential` across two different environments:

Reviewer: ```suggestion
The [Azure Identity library](/dotnet/api/azure.identity?view=azure-dotnet&preserve-view=true) provides various *credentials*&mdash;implementations of `TokenCredential` adapted to supporting different scenarios and Microsoft Entra authentication flows. Since managed identity is unavailable when running locally, the steps ahead demonstrate which credential to use in which scenario:
```

> - **Local dev environment**: During **local development only**, use a class called `DefaultAzureCredential` for an opinionated, preconfigured chain of credentials. `DefaultAzureCredential` discovers user credentials from your local tooling or IDE, such as the Azure CLI or Visual Studio. It also provides flexibility and convenience for retries, wait times for responses, and support for multiple authentication options. Visit the [Authenticate to Azure services during local development](/dotnet/azure/sdk/authentication/local-development-dev-accounts) article to learn more.

Reviewer: ```suggestion
- **Local dev environment**: During **local development only**, use a class called [DefaultAzureCredential](/dotnet/azure/sdk/authentication/credential-chains#defaultazurecredential-overview) for an opinionated, preconfigured chain of credentials. `DefaultAzureCredential` discovers user credentials from your local tooling or IDE, such as the Azure CLI or Visual Studio. It also provides flexibility and convenience for retries, wait times for responses, and support for multiple authentication options. Visit the [Authenticate to Azure services during local development](/dotnet/azure/sdk/authentication/local-development-dev-accounts) article to learn more.
```

> - **Azure-hosted apps**: When your app is running in Azure, use `ManagedIdentityCredential` to safely discover the managed identity configured for your app. Specifying this exact type of credential prevents other available credentials from being picked up unexpectedly.

Reviewer: ```suggestion
- **Azure-hosted apps**: When your app is running in Azure, use [ManagedIdentityCredential](/dotnet/api/azure.identity.managedidentitycredential?view=azure-dotnet&preserve-view=true) to safely discover the managed identity configured for your app. Specifying this exact type of credential prevents other available credentials from being picked up unexpectedly.
```

> - **Azure-hosted apps**: When your app is running in Azure, use a `ManagedIdentityCredential` to safely discover the managed identity configured for your app. Specifying this exact type of credential prevents other available credentials from being picked up unexpectedly.

Reviewer: ```suggestion
- **Azure-hosted apps**: When your app is running in Azure, use `ManagedIdentityCredential` to safely discover the managed identity configured for your app. Specifying this exact type of credential prevents other available credentials from being picked up unexpectedly.
```
